### PR TITLE
CLOUDP-263186: Add Logic to retrieve the FOASMetadata from APIx s3 bucket

### PR DESCRIPTION
**Pull request by andreaangiolillo**
This PR adds the logic to retrieve the FOASMetadata from APIx s3 bucket.

I tested my changes in a fork: https://github.com/andreaangiolillo/openapi-test/actions/runs/10096226482

Screenshot 2024-07-25 at 15 46 46

# Next Steps
I plan to move the FOAS generation logic to another workflow that is called from the main one similar to the validations.

### CODE REVIEW

File: .github/workflows/release-spec-prod.yml [hunk 1]
```diff
@@ -7,20 +7,80 @@ permissions:
...
+        run: make build
+      - name: Retrieve Specs
+        working-directory: tools/cli/bin
+        env:
+          AWS_DEFAULT_REGION: ${{vars.AWS_DEFAULT_REGION}}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID_PROD }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY_PROD }}
+          S3_BUCKET: ${{ vars.S3_BUCKET_PROD }}
+        run: |
+          aws s3 cp s3://"${S3_BUCKET}"/openapi/foas-metadata.json foas-metadata.json
```
Comment (blva) [inline .github/workflows/release-spec-prod.yml:35]: [nit] can this live in a .sh file so that it can be linted?

Comment (andreaangiolillo) [reply .github/workflows/release-spec-prod.yml:35]: yes, but it is already linted by the `code_health`worfkflow at the step `actionlint`

Comment (andreaangiolillo) [reply .github/workflows/release-spec-prod.yml:35]: happy to move it to a script in a follow up PR if you think would help 👍

Comment (blva) [reply .github/workflows/release-spec-prod.yml:35]: based on evergreen life experience, I'd rather have as much logic as possible out of actions and into something we can unit test, and actions would only pass the secrets/variables. that way we can see how our changes affect those flows ahead of time. not a strong opinion, up to you. e.g. build a foascli release command or a releasecli to do the foas metadata handling (think that in the future we can change that format, or s3 bucket structure, or how it works for each branch)

Comment (andreaangiolillo) [reply .github/workflows/release-spec-prod.yml:35]: addressed this comment in https://github.com/mongodb/openapi/pull/114

**Pull request merged**